app/components/form.tsx

```diff
@@ -373,90 +378,93 @@ export const Select = ({
-                {hint && <DisabledMessageIcon message={hint} />}
-              </Box>
-            );
-          }}
-        >
-          {sortedOptions.map((opt) => {
-            if (!opt.value && opt.type !== "group") {
-              return null;
-            }
-
-            return opt.type === "group" ? (
-              opt.label && (
-                <ListSubheader key={opt.label}>{opt.label}</ListSubheader>
-              )
-            ) : (
-              <MenuItem
-                key={opt.value}
-                disabled={opt.disabled}
-                value={opt.value ?? undefined}
-                sx={{
-                  display: "flex",
-                  alignItems: "center",
-                  "&.Mui-disabled": {
-                    opacity: 1,
-
-                    "&:hover": {
-                      backgroundColor: "transparent",
+                  <InputLabel
+                    sx={{
+                      typography: "body2",
+                      color: "secondary.active",
+                      pointerEvents: "none",
+                    }}
+                  >
+                    {selectedOption.label}
+                  </InputLabel>
+                  {hint && <DisabledMessageIcon message={hint} />}
+                </Box>
+              );
+            }}
+          >
+            {sortedOptions.map((opt) => {
+              if (!opt.value && opt.type !== "group") {
+                return null;
+              }
+
+              return opt.type === "group" ? (
+                opt.label && (
+                  <ListSubheader key={opt.label}>{opt.label}</ListSubheader>
+                )
+              ) : (
+                <MenuItem
+                  key={opt.value}
+                  disabled={opt.disabled}
+                  value={opt.value ?? undefined}
+                  sx={{
+                    display: "flex",
+                    alignItems: "center",
+                    "&.Mui-disabled": {
+                      opacity: 1,
+
+                      "&:hover": {
+                        backgroundColor: "transparent",
+                      },
                     },
-                  },
-                }}
-              >
-                <span style={{ opacity: opt.disabled ? 0.38 : 1 }}>
-                  {opt.label}
-                </span>
-                {opt.disabledMessage && (
-                  <DisabledMessageIcon message={opt.disabledMessage} />
-                )}
-              </MenuItem>
-            );
-          })}
-        </MUISelect>
+                  }}
+                >
+                  <span style={{ opacity: opt.disabled ? 0.38 : 1 }}>
+                    {opt.label}
+                  </span>
+                  {opt.disabledMessage && (
+                    <DisabledMessageIcon message={opt.disabledMessage} />
+                  )}
+                </MenuItem>
+              );
+            })}
+          </MUISelect>
+          {sideControls}
```

Comment:
I have the impression that sideControls should be wrapped into a flexShrink: 0 container.

<img width="349" alt="image" src="https://github.com/visualize-admin/visualization-tool/assets/465582/e2ce270d-4437-4fff-8859-d436e334187d">

Created from : https://visualization-tool-git-feat-most-recent-x-axis-ixt1.vercel.app/en/browse?dataset=https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Fnfi%2Fnfi_T-changes%2Fcube%2F2024-1